crates/subspace-core-primitives/Cargo.toml

```diff
@@ -17,6 +17,7 @@ parity-scale-codec = { version = "2.3.0", default-features = false, features = [
 scale-info = { version = "1.0", default-features = false, features = ["derive"] }
 serde = { version = "1.0.136", optional = true, features = ["derive"] }
 serde_arrays = { version = "0.1", optional = true }
+sp-core = { version = "5.0.0", git = "https://github.com/paritytech/substrate", rev = "e6def65920d30029e42d498cb07cec5dd433b927" }
```

Comment:
This crate is supposed to be independent from anything substrate-related, so this dependency should be avoided.

You only really need this for an external crate for CLI purposes, so you can just customize string parsing there instead: https://github.com/clap-rs/clap/blob/v3.1.3/examples/derive_ref/README.md#arg-attributes